blaze/compute/sql.py

```diff
@@ -585,6 +592,17 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return getattr(sa.func, expr.attr)(data).label(expr._name)
```

Comment:
if there was a way to get the dialect out from an arbitrary sqlalchemy expression then we could support everything